notebooks/236-raft-optimization/236-raft-optimization.ipynb

```diff
@@ -0,0 +1,570 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Convert and Optimize RAFT with OpenVINO™\n",
+    "\n",
+    "The Recurrent All-Pairs Field Transforms (RAFT) [repository](https://github.com/princeton-vl/RAFT) is a new deep network architecture for optical flow. \n",
+    "\n",
+    "RAFT extracts perpixel features, builds multi-scale 4D correlation volumes for all pairs of pixels, and iteratively updates a flow field through a recurrent unit that performs lookups on the correlation volumes. RAFT achieves stateof-the-art performance.\n",
+    "\n",
+    "This tutorial demonstrates step-by-step instructions on how to run and optimize RAFT with OpenVINO based on the [ONNX-RAFT-Optical-Flow-Estimation](https://github.com/ibaiGorordo/ONNX-RAFT-Optical-Flow-Estimation/tree/main). \n",
+    "\n",
+    "The tutorial consists of the following steps:\n",
+    "- Download ONNX model\n",
```

Comment:
what the source of onnx model? Who and how convert it?
Is it possible to demonstrate how to correctly convert pytorch model to onnx in notebook? I think it maybe useful if somebody wants run different checkpoint or own pretrained model

Reply:
Hi eaidova,
Thanks for the reply. I will add the [repo link](https://github.com/ibaiGorordo/ONNX-RAFT-Optical-Flow-Estimation/tree/main) into the readme file. 
[PINTO_model_zoo](https://github.com/PINTO0309/PINTO_model_zoo/tree/04f86a550e2ce1c5fb70cfafdfdeb568700f61c2/252_RAFT) converted the model.

I directly converted model from the raft official repo, yet failed with benchmark_app ov22.3. 

CPU failed
RuntimeError: There should be only one instance of RegistersPool per thread
GPU failed
RuntimeError: Invalid dimensions size(7) for gpu tensor

It works with low performance on OV23.0(20230314)

Test on OV23.0

(base) root@worker44:/home/worker/ov23.0/openvino/install_gpu_onednn/samples/cpp# ./intel64/Release/benchmark_app -m /home/worker/ys/raft_small_iter20_480x640.onnx -d CPU

[ INFO ] Throughput:          13.42 FPS

(base) root@worker44:/home/worker/ov23.0/openvino/install_gpu_onednn/samples/cpp# ./intel64/Release/benchmark_app -m /home/worker/ys/raft_small_iter20_480x640.onnx -d GPU -exec_graph_path /home/worker/ys/raft_small_iter20_480x640.xml

[ INFO ] Throughput:          13.42 FPS

Reply:
ok, maybe then we can postpone this, thank you for sharing